ppl/zqd/auth.go

```diff
@@ -0,0 +1,145 @@
+package zqd
+
+import (
+	"context"
+	"crypto/rsa"
+	"errors"
+	"flag"
+	"fmt"
+	"net/http"
+
+	jwtmiddleware "github.com/auth0/go-jwt-middleware"
+	"github.com/brimsec/zq/pkg/fs"
+	"github.com/dgrijalva/jwt-go"
+	"github.com/gorilla/mux"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"go.uber.org/zap"
+)
+
+const (
+	// authTokenContextValue is the string the jwtmiddleware library uses to
+	// store the validated & parsed JWT in a request's context.
+	authTokenContextValue = "zqd-core-auth-token"
+
+	tenantIDClaim = "brim_tenant_id"
+	userIDClaim   = "brim_user_id"
+)
+
+type AuthConfig struct {
+	Enabled  bool
+	Audience string
+	Issuer   string
+	JWKSPath string
+}
+
+func (c *AuthConfig) SetFlags(fs *flag.FlagSet) {
+	fs.BoolVar(&c.Enabled, "auth.enabled", false, "enable authentication checks")
+	fs.StringVar(&c.Audience, "auth.audience", "", "required JWT audience claim")
+	fs.StringVar(&c.Issuer, "auth.issuer", "", "required JWT issuer claim")
+	fs.StringVar(&c.JWKSPath, "auth.jwkspath", "", "path to JSON Web Key Set file")
+}
```

Comment:
OK, now that I see this, I can make changes in #1695 so the "worker" flags follow the same pattern.